basics.py

```diff
@@ -1,15 +1,18 @@
 import spacy
 import os
 
+#KCE: add output type hint
 def file_generator(dir: str):
-    for filename in sorted(os.listdir(dir)):
+    for filename in sorted(os.listdir(dir)): # KCE: Why sort the dir?
```

Comment:
Only as convenience; when printing progress on yield, I can use the filenames as a progress indicator.